proof_ipa.go

```diff
@@ -353,6 +354,7 @@ func TreeFromProof(proof *Proof, rootC *Point) (VerkleNode, error) {
 				continue
 			}
 
+			// TODO(jsign): this if will never be true.
 			if bytes.Equal(k, info[string(p)].stem) {
 				values[k[31]] = proof.Values[i]
 			}
```

Comment:
@gballet, doesn't seem that this `if` will ever be true. We're comparing a `k` (32 byte key) with a stem which is 31 byte.